File: gcc/testsuite/c-c++-common/gomp/flush-2.c
int a, b;

void
foo (void)
{
  #pragma omp flush
  #pragma omp flush (a, b)
  #pragma omp flush acquire
  #pragma omp flush release
  #pragma omp flush acq_rel
  #pragma omp flush relaxed		/* { dg-error "expected 'acq_rel', 'release' or 'acquire'" } */
  #pragma omp flush seq_cst		/* { dg-error "expected 'acq_rel', 'release' or 'acquire'" } */
  #pragma omp flush foobar		/* { dg-error "expected 'acq_rel', 'release' or 'acquire'" } */
  #pragma omp flush acquire (a, b)	/* { dg-error "'flush' list specified together with memory order clause" } */
  #pragma omp flush release (a, b)	/* { dg-error "'flush' list specified together with memory order clause" } */
  #pragma omp flush acq_rel (a, b)	/* { dg-error "'flush' list specified together with memory order clause" } */
}

File: gcc/testsuite/gfortran.dg/gomp/flush-2.f90
module m
  integer :: a, b
end module m

subroutine foo (void)
  use m
  !$omp flush
  !$omp flush (a, b)
  !$omp flush acquire
  !$omp flush release
  !$omp flush acq_rel
  !$omp flush seq_cst
  !$omp flush relaxed		! { dg-error "Expected SEQ_CST, AQC_REL, RELEASE, or ACQUIRE" }
  !$omp flush foobar		! { dg-error "Expected SEQ_CST, AQC_REL, RELEASE, or ACQUIRE" }
  !$omp flush acquire (a, b)	! { dg-error "List specified together with memory order clause in FLUSH directive" }
  !$omp flush release (a, b)	! { dg-error "List specified together with memory order clause in FLUSH directive" }
  !$omp flush acq_rel (a, b)	! { dg-error "List specified together with memory order clause in FLUSH directive" }
  !$omp flush seq_cst (a, b)	! { dg-error "List specified together with memory order clause in FLUSH directive" }
  end
